From 4f2a318cd79be7d15aa83f93526fdf87dab8d2d7 Mon Sep 17 00:00:00 2001
From: Jesse Chan <jc@lineageos.org>
Date: Fri, 3 Mar 2017 20:42:40 +0800
Subject: [PATCH 13/15] stagefright: allow multiple custom OMXPlugins

* Separated by comma(,)
* Example: media.sf.omx-plugin=libffmpeg_omx.so,libsomxcore.so

Change-Id: I15556a48df282b01f54ca864317eafff5468e739
Signed-off-by: Jesse Chan <jc@lineageos.org>
---
 media/libstagefright/omx/OMXMaster.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/omx/OMXMaster.cpp b/media/libstagefright/omx/OMXMaster.cpp
index 7c93f2bff1..b2109fd547 100644
--- a/media/libstagefright/omx/OMXMaster.cpp
+++ b/media/libstagefright/omx/OMXMaster.cpp
@@ -71,8 +71,13 @@ void OMXMaster::addPlatformPlugin() {
 
 void OMXMaster::addUserPlugin() {
     char plugin[PROPERTY_VALUE_MAX];
+    char *each_plugin;
     if (property_get("media.sf.omx-plugin", plugin, NULL)) {
-        addPlugin(plugin);
+        each_plugin = strtok(plugin, ",");
+        while (each_plugin != NULL) {
+            addPlugin(each_plugin);
+            each_plugin = strtok(NULL, ",");
+        }
     }
 }
 
-- 
2.21.0

